kie-api/src/main/java/org/kie/api/internal/assembler/KieAssemblerService.java

```diff
@@ -37,7 +37,9 @@ default void addResourcesBeforeRules(Object kbuilder, Collection<ResourceWithCon
         }
     }
 
-    default void addResourceAfterRules(Object kbuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception { }
+    default void addResourceAfterRules(Object kbuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception {
+        addResource(kbuilder, resource, type, configuration);
+    }
```

Comment:
Doesn't this introduce a cycle?
* addResource calls addResourceAfterRules
* addResourceAfterRules calls addResource

Reply:
@MarianMacik 
fixed